sunpy/map/sources/tests/test_xrt_source.py

```diff
@@ -13,45 +13,58 @@
 
 
 @pytest.fixture
-def xrt():
+def xrt_map():
     return get_dummy_map_from_header(get_test_filepath("HinodeXRT.header"))
 
 
-# XRT Tests
-def test_fitstoXRT(xrt):
+def test_fits_to_xrt(xrt_map):
     """Tests the creation of XRTMap using FITS."""
-    assert isinstance(xrt, XRTMap)
+    assert isinstance(xrt_map, XRTMap)
 
 
-def test_is_datasource_for(xrt):
+def test_is_datasource_for(xrt_map):
     """Test the is_datasource_for method of XRTMap.
     Note that header data to be provided as an argument
     can be a MetaDict object."""
-    assert xrt.is_datasource_for(xrt.data, xrt.meta)
+    assert xrt_map.is_datasource_for(xrt_map.data, xrt_map.meta)
 
 
-def test_observatory(xrt):
+def test_observatory(xrt_map):
     """Tests the observatory property of the XRTMap object."""
-    assert xrt.observatory == "Hinode"
+    assert xrt_map.observatory == "Hinode"
 
 
-def test_measurement(xrt):
+def test_measurement(xrt_map):
     """Tests the measurement property of the XRTMap object."""
-    measurement = xrt.filter_wheel1_measurements[5].replace("_", " ")
-    measurement += '-' + xrt.filter_wheel2_measurements[1].replace("_", " ")
-    assert xrt.measurement == measurement
+    measurement = xrt_map.filter_wheel1_measurements[5].replace("_", " ")
+    measurement += '-' + xrt_map.filter_wheel2_measurements[1].replace("_", " ")
+    assert xrt_map.measurement == measurement
 
 
-def test_wheel_measurements(xrt):
+def test_level_number(xrt_map):
+    assert xrt_map.processing_level == 0
+
+
+def test_heliographic_longitude(xrt_map):
+    with pytest.warns(SunpyMetadataWarning, match='Missing metadata for observer'):
```

Comment:
I thought the point was to get rid of these warnings?

Reply:
We use a level 0 header from what I can see and that will raise this warning. 

In a level 1 header (which I still need to source), it should not raise this warning.

Reply:
Are we not trying to get rid of the warning for the L0 header too?

Reply:
I think we would need to calculate solar_b0 if its missing for the level 0 headers but I think people are meant to use the level 1 files.

Reply:
I am slightly opposed to suppressing the warning for L0 files

Reply:
Why? Doesn't the same logic for suppressing the warning for L1 files apply for L0 files?

Reply:
Here's the distinction that I'm drawing.  When generating the L1 files, the XRT team has decided that specifying `SOLAR_B0` and `DSUN_OBS` is sufficient observer metadata, and that the omission of `HGLN_OBS` shall be interpreted as `HGLN_OBS = 0`.  This PR implements that specific handling of an intentionally omitted keyword.  In contrast, the L0 files are lacking observer metadata entirely, so to me are still a bit raw rather than "analysis-ready".  Since our assuming that the observer is at Earth center injects information rather than fixes how we interpret existing information, it feels more dramatic and still deserving of a warning, even if the answer turns out to be the same as what the XRT team does when promoting L0 to L1.